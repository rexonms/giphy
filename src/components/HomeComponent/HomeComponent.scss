.container {

}
.searchBox {
  max-width: 80%;
  padding-top: 40px;
  margin: 0 auto 20px auto;
}
.filters {
  min-height: 50px;
  margin-bottom: 40px;
  background: #030303;
}
.gifs {
  min-height: 500px;
  background: #010101;
  padding: $defaultPadding;
}
