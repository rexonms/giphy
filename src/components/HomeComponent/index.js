import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Styles from './HomeComponent.scss';
import SearchBox from '../lib/SearchBox';
import MosaicContainer from '../lib/MosaicContainer';

const propTypes = {
  searchBox: PropTypes.shape({}).isRequired,
  giphy: PropTypes.shape({
    data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  }).isRequired,
  getSearchedGifs: PropTypes.func.isRequired,
};

const defaultProps = {};

class HomeContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  onClickHandler() {
    this.setState({
      bulbState: !this.state.bulbState,
    });
  }

  render() {
    const { searchBox } = this.props;
    return (
      <div className={Styles.container}>
        <div className={Styles.searchBox}>
          <SearchBox
            placeholder={searchBox.placeholder}
            buttonText={searchBox.buttonText}
            onSearchHandler={keyword => this.props.getSearchedGifs(keyword)}
          />
        </div>
        <div className={Styles.filters} />
        <div className={Styles.result}>
          <MosaicContainer gifs={this.props.giphy.data} />
        </div>
      </div>
    );
  }
}

HomeContainer.propTypes = propTypes;
HomeContainer.defaultProps = defaultProps;

export default HomeContainer;
