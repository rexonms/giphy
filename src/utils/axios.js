import axios from 'axios';

export const foo = 'bazz';
export const getDataFromGiphy = async (url) => {
  const response = await axios.get(url);
  return response;
};

export const giphyDefaults = {
  baseURL: process.env.REACT_APP_GIPHY_BASE_URL || 'https://api.giphy.com',
  apiKey: process.env.REACT_APP_GIPHY_API_KEY || 'eazZw1kI9vdWbXRls4gVKo4LCGZQtoLb',
  limit: 25,
  rating: 'g',
  offset: 0,
  endpoints: {
    trending: 'trending',
    search: 'search',
  },
};
