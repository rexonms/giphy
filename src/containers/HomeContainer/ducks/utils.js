import { giphyDefaults } from '../../../utils/axios';

export const getUrl = ({ home, endpoint }) => {
  const base = `${giphyDefaults.baseURL}/v1/gifs/${endpoint.name}`;
  const api = `api_key=${giphyDefaults.apiKey}`;
  const rating = `rating=${giphyDefaults.rating}`;
  const url = `${base}?${api}&${rating}`;
  let limit = `limit=${giphyDefaults.limit}`;
  if (endpoint.name === giphyDefaults.endpoints.trending) {
    // check if there is any
    if (home.giphy) {
      limit = `limit=${home.giphy.pagination.count + giphyDefaults.limit}`;
    }
    return `${url}&${limit}`;
  }
  const offsetValue = endpoint.query === home.giphy.endpoint.query ?
    home.giphy.pagination.offset + giphyDefaults.limit + 1 : 0;
  const offset = `offset=${offsetValue}`;
  return `${url}&${limit}&${offset}&q=${endpoint.query}`;
};

export const extractResponseInfo = (data, endpoint) => (
  {
    endpoint,
    pagination: data.pagination,
    data: data.data.map(d => (
      {
        id: d.id,
        title: d.title,
        rating: d.rating,
        user: {
          avatarUrl: d.user ? d.user.avatar_url : 'null',
        },
        image: {
          url: d.images.fixed_height_downsampled,
        },
      }
    )),
  }
);
