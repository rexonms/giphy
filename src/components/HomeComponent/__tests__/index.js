import React from 'react';
import { shallow } from 'enzyme';

import HomeComponent from './../index';
import { giphyDefaults } from '../../../utils/axios';

describe('Welcome Component', () => {
  let props;
  let shallowHomeComponent;
  const homeComponent = () => {
    if (!shallowHomeComponent) {
      shallowHomeComponent = shallow(<HomeComponent {...props} />);
    }
    return shallowHomeComponent;
  };
  beforeEach(() => {
    props = {
      home: {
        searchBox: {
          value: '',
          buttonText: 'Search',
          placeholder: 'Start Typing',
          onSearchHandler: () => {},
        },
        giphy: {
          endpoint: {
            name: giphyDefaults.endpoints.trending,
            query: 'foo',
          },
          pagination: {
            total_count: 56795,
            count: 3,
            offset: 0,
          },
          data: [
            {
              id: 'xUOrwbGjODhnRe0ZTW',
              title: 'creepy thats my fetish GIF by NETFLIX',
              rating: 'g',
              user: {
                avatarUrl: 'https://media.giphy.com/avatars/netflix/bAiLTTOSCowX.png',
              },
              image: {
                url: {
                  url: 'https://media3.giphy.com/media/xUOrwbGjODhnRe0ZTW/200_d.gif',
                  width: '200',
                  height: '200',
                  size: '152944',
                  webp: 'https://media3.giphy.com/media/xUOrwbGjODhnRe0ZTW/200_d.webp',
                  webp_size: '49912',
                },
              },
            },
            {
              id: 'l2JJyDYEX1tXFmCd2',
              title: 'julia louis dreyfus happy dance GIF by CraveTV',
              rating: 'g',
              user: {
                avatarUrl: 'https://media.giphy.com/avatars/cravetvcanada/O77NNU18q3fH.jpg',
              },
              image: {
                url: {
                  url: 'https://media0.giphy.com/media/l2JJyDYEX1tXFmCd2/200_d.gif',
                  width: '347',
                  height: '200',
                  size: '185842',
                  webp: 'https://media0.giphy.com/media/l2JJyDYEX1tXFmCd2/200_d.webp',
                  webp_size: '58368',
                },
              },
            },
            {
              id: 'l0HlIk8EALDCUHCA8',
              title: 'high school musical summer GIF by Refinery 29 GIFs',
              rating: 'g',
              user: {
                avatarUrl: 'https://media0.giphy.com/avatars/refinery29gifs/LLnRhUU66BIl.png',
              },
              image: {
                url: {
                  url: 'https://media3.giphy.com/media/l0HlIk8EALDCUHCA8/200_d.gif',
                  width: '200',
                  height: '200',
                  size: '79257',
                  webp: 'https://media3.giphy.com/media/l0HlIk8EALDCUHCA8/200_d.webp',
                  webp_size: '51750',
                },
              },
            },
          ],
        },
      },
      getGifs: jest.fn(),
    };
    shallowHomeComponent = undefined;
  });

  it('should render previous snapshot', () => {
    expect(homeComponent()).toMatchSnapshot();
  });
  it('should render with proper className', () => {
    expect(homeComponent().find('.container').exists()).toBe(true);
    expect(homeComponent().find('.searchBox').exists()).toBe(true);
    expect(homeComponent().find('.gifs').exists()).toBe(true);
  });
  it('should have child components', () => {
    expect(homeComponent().find('SearchBox').exists()).toBe(true);
    expect(homeComponent().find('MosaicComponent').exists()).toBe(true);
  });
});
