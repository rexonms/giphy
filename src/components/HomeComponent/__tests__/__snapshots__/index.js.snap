// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component should render previous snapshot 1`] = `
<div
  className="container"
>
  <div
    className="searchBox"
  >
    <SearchBox
      buttonText="Search"
      onSearchHandler={[Function]}
      placeholder="Start Typing"
      value=""
    />
  </div>
  <div
    className="filters"
  />
  <div
    className="gifs"
  >
    <MosaicComponent
      action={[Function]}
      endpoint={
        Object {
          "name": "trending",
          "query": "foo",
        }
      }
      gifs={
        Array [
          Object {
            "id": "xUOrwbGjODhnRe0ZTW",
            "image": Object {
              "url": Object {
                "height": "200",
                "size": "152944",
                "url": "https://media3.giphy.com/media/xUOrwbGjODhnRe0ZTW/200_d.gif",
                "webp": "https://media3.giphy.com/media/xUOrwbGjODhnRe0ZTW/200_d.webp",
                "webp_size": "49912",
                "width": "200",
              },
            },
            "rating": "g",
            "title": "creepy thats my fetish GIF by NETFLIX",
            "user": Object {
              "avatarUrl": "https://media.giphy.com/avatars/netflix/bAiLTTOSCowX.png",
            },
          },
          Object {
            "id": "l2JJyDYEX1tXFmCd2",
            "image": Object {
              "url": Object {
                "height": "200",
                "size": "185842",
                "url": "https://media0.giphy.com/media/l2JJyDYEX1tXFmCd2/200_d.gif",
                "webp": "https://media0.giphy.com/media/l2JJyDYEX1tXFmCd2/200_d.webp",
                "webp_size": "58368",
                "width": "347",
              },
            },
            "rating": "g",
            "title": "julia louis dreyfus happy dance GIF by CraveTV",
            "user": Object {
              "avatarUrl": "https://media.giphy.com/avatars/cravetvcanada/O77NNU18q3fH.jpg",
            },
          },
          Object {
            "id": "l0HlIk8EALDCUHCA8",
            "image": Object {
              "url": Object {
                "height": "200",
                "size": "79257",
                "url": "https://media3.giphy.com/media/l0HlIk8EALDCUHCA8/200_d.gif",
                "webp": "https://media3.giphy.com/media/l0HlIk8EALDCUHCA8/200_d.webp",
                "webp_size": "51750",
                "width": "200",
              },
            },
            "rating": "g",
            "title": "high school musical summer GIF by Refinery 29 GIFs",
            "user": Object {
              "avatarUrl": "https://media0.giphy.com/avatars/refinery29gifs/LLnRhUU66BIl.png",
            },
          },
        ]
      }
    />
  </div>
</div>
`;
